### PR TITLE
enable NOK as a currency. 

### PR DESCRIPTION
**Pull request by akashpatl**
Still needs exclusivity filters for hindering cross market redemptions

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/campaign/service/ReferralServiceImpl.kt [hunk 1]
```diff
@@ -44,6 +44,10 @@ constructor(
     val referralCampaign = member.referralCampaign
       ?: throw NullPointerException("No ReferralCampaign for campaignOwner ${member.ownerId}")
 
+    if (referralCampaign.incentive.currency ==  null) {
+      throw java.lang.NullPointerException("Member does not have a currency on ReferralCampaign: {$memberId}")
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/productPricing/campaign/service/ReferralServiceImpl.kt]: Kotlin npe is also fine :) 
```suggestion
      throw NullPointerException("Member does not have a currency on ReferralCampaign: {$memberId}")
```

File: src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt [hunk 2]
```diff
@@ -138,7 +141,7 @@ class CampaignServiceImpl(
         }
       }
     }
-
+    //TODO: Add a check between currencies to see if they are the same else throw exception.
     val discount = FastMoney.of(totalDiscount.max(BigDecimal.ZERO), ReferralServiceImpl.SEK)
     val netPrice = MonetaryFunctions.max()
       .apply(grossPrice.subtract(discount), FastMoney.of(0, ReferralServiceImpl.SEK))
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt]: Yes lets fix this todo set you could just `discount` and `netPrice` currency to  preferredCurrency

File: src/main/java/com/hedvig/productPricing/campaign/service/CampaignBillingServiceImpl.kt [hunk 3]
```diff
@@ -152,7 +160,15 @@ class CampaignBillingServiceImpl(
...
+  private fun getCurrencyFromProducts(memberId: String, productSubscription: List<BillableProductSubscription>): CurrencyUnit {
+    val distinctCurrencies = productSubscription
+      .map { o -> o.subscription.currency }.toSet()
+    require(distinctCurrencies.size == 1) {"Multiple currencies on product billing for member id: {$memberId}. Currencies are: {$distinctCurrencies}"}
+    return distinctCurrencies.first()
+  }
+
   companion object {
     const val SEK: String = "SEK"
+    const val NOK: String = "NOK"
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/campaign/service/CampaignBillingServiceImpl.kt]: Do we use them now?

File: src/main/java/com/hedvig/productPricing/service/query/CampaignStatusEventListener.kt [hunk 4]
```diff
@@ -32,15 +32,16 @@ class CampaignStatusEventListener(
...
+    domainService.notifyMemberSigned(e.holderMemberId, timestamp, e.agreement.getPreferredCurrency().currencyCode)
 
     domainService.notifyMemberInsuranceFirstActivation(e.holderMemberId, DateUtil.toStockholmInstant(e.agreement.from!!.atStartOfDay()))
   }
 
   @EventHandler
-  fun on(e: ContractWithPendingAgreementCreated<AgreementStereotype>, @Timestamp timestamp: Instant) {
-    domainService.notifyMemberSigned(e.holderMemberId, timestamp)
+  fun on(e: ContractWithPendingAgreementCreated<BaseAgreement>, @Timestamp timestamp: Instant) {
+    domainService.notifyMemberSigned(e.holderMemberId, timestamp, e.agreement.getPreferredCurrency().currencyCode)
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/service/query/CampaignStatusEventListener.kt]: SWEET

File: bin/docker_push.sh [hunk 5]
```diff
@@ -2,7 +2,7 @@
 # Push only if it's not a pull request
 if [ -z "$TRAVIS_PULL_REQUEST" ] || [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
   # Push only if we're testing the master branch
-  if [ "$TRAVIS_BRANCH" == "master" ]; then
```
Comment (meletios) [inline bin/docker_push.sh]: remove this

File: bin/kubernetes_deploy.sh [hunk 6]
```diff
@@ -3,7 +3,7 @@
 # Push only if it's not a pull request
 if [[ -z "$TRAVIS_PULL_REQUEST" ]] || [[ "$TRAVIS_PULL_REQUEST" == "false" ]]; then
   # Push only if we're testing the master branch
-  if [[ "$TRAVIS_BRANCH" == "master" ]] ; then
```
Comment (meletios) [inline bin/kubernetes_deploy.sh]: we dont need this

File: bin/docker_push.sh [hunk 7]
```diff
@@ -2,7 +2,7 @@
...
+  if [ "$TRAVIS_BRANCH" == "master" ] || [ "$TRAVIS_BRANCH" == "feature/norway-campagins-parity-v2" ]; then
```
Comment (FredrikAreschoug) [inline bin/docker_push.sh]:
```suggestion
  if [ "$TRAVIS_BRANCH" == "master" ]; then
```

File: bin/kubernetes_deploy.sh [hunk 8]
```diff
@@ -3,7 +3,7 @@
...
+  if [[ "$TRAVIS_BRANCH" == "master" ]] || [[ "$TRAVIS_BRANCH" == "feature/norway-campagins-parity-v2" ]] ; then
```
Comment (FredrikAreschoug) [inline bin/kubernetes_deploy.sh]:
```suggestion
  if [[ "$TRAVIS_BRANCH" == "master" ]] ; then
```

File: bin/docker_push.sh (hunk 7 above)
Comment (meletios) [inline bin/docker_push.sh]: We dont need this

File: bin/kubernetes_deploy.sh (hunk 8 above)
Comment (meletios) [inline bin/kubernetes_deploy.sh]: and this

**Pull request merged**